src/wmtk/TetMeshOperationExecutor.cpp

```diff
@@ -828,18 +754,34 @@ Tuple TetMesh::TetMeshOperationExecutor::collapse_edge()
             auto tt = tt_accessor.index_access().vector_attribute(t_ear_1);
             auto tf = tf_accessor.index_access().vector_attribute(t_ear_1);
             auto te = te_accessor.index_access().vector_attribute(t_ear_1);
+            auto tv = tv_accessor.index_access().vector_attribute(t_ear_1);
 
             for (int k = 0; k < 4; k++) {
                 if (tf(k) == f_ear_1) {
                     assert(tt(k) == t_old);
                     tf(k) = f_ear_2;
                     tt(k) = t_ear_2;
+
+                    // for return tuple
+                    if (return_flag && return_tid == t_ear_1) {
```

Comment:
I don't quite understand why this code needs to be in the for-loop. The if-statement is `true` or `false` in every iteration, or am I missing something? `return_flag`, `return_tid`, and `t_ear_1` never change in the for-loop.

Is it because of the outer if-statement? In that case, I feel like it would be cleaner to iterate through `k` once more and do the return tuple stuff independently of the current loop.